src/components/PluginInstallationsPercentageChart.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useRef } from 'react'
+import * as echarts from 'echarts'
+import dayjs from 'dayjs'
+import { PluginChartProps } from '../data/plugins'
+
+const PluginInstallationsPercentageChart: React.FC<PluginChartProps> = ({ data }) => {
+    const chartRef = useRef(null)
+
+    useEffect(() => {
```

Comment:
Think this could be either a `useMemo` or a `useCallback`, but not a `useEffect`, because we are not dealing with React side effects here: https://www.w3schools.com/react/react_useeffect.asp

Reply:
I think we are using react side effects in this instance.

Specifically regarding the resizing on DOM manipulation  window.addEventListener('resize', handleResize)

and updating the echarts instance based on the data prop. 

let me know if we can discuss further.

Reply:
Yes and no, since the top part are definitely not side effects but the bottom part is.

Reply:
Yeah, we should talk about this. It seems like there is too many things packed to the same `useEffect` hook...